Dockerfile

```
@@ -13,14 +13,11 @@ WORKDIR /app
 # Copy source files
-COPY requirements.txt requirements-e2e.txt setup.py ./
-COPY README.md ./
+COPY pyproject.toml MANIFEST.in setup.py README.md LICENSE ./
 COPY src/ ./src/
 
-# Install dependencies and application package with E2EE support (consolidated)
-RUN python -m pip install --no-cache-dir --timeout=300 --retries=3 -r requirements.txt \
-    && python -m pip install --no-cache-dir --timeout=300 --retries=3 -r requirements-e2e.txt \
-    && python -m pip install --no-cache-dir --no-deps .
+# Install dependencies and application package with E2EE support
+RUN python -m pip install --no-cache-dir --timeout=300 --retries=3 ".[e2e]"
```

Hadolint (2.14.0)

[warning] Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

(DL3013)